router/router.go

```diff
@@ -0,0 +1,57 @@
+package router
+
+import (
+	"github.com/99designs/gqlgen/graphql/handler"
+	"github.com/99designs/gqlgen/graphql/playground"
+	"github.com/gin-gonic/gin"
+	"github.com/speedrun-website/leaderboard-backend/graph"
+	"github.com/speedrun-website/leaderboard-backend/graph/generated"
+	handlers "github.com/speedrun-website/leaderboard-backend/handlers"
+	"github.com/speedrun-website/leaderboard-backend/middleware"
+	"github.com/speedrun-website/leaderboard-backend/validators"
+)
+
+// Defining the Graphql handler
+func graphqlHandler() gin.HandlerFunc {
+	// NewExecutableSchema and Config are in the generated.go file
+	// Resolver is in the resolver.go file
+	h := handler.NewDefaultServer(generated.NewExecutableSchema(generated.Config{Resolvers: &graph.Resolver{}}))
+
+	return func(c *gin.Context) {
+		h.ServeHTTP(c.Writer, c.Request)
+	}
+}
+
+// Defining the Playground handler
+func playgroundHandler() gin.HandlerFunc {
+	h := playground.Handler("GraphQL", "/query")
+
+	return func(c *gin.Context) {
+		h.ServeHTTP(c.Writer, c.Request)
+	}
+}
+
+func InitRoutes(router *gin.Engine) {
+	router.GET("/playground", playgroundHandler())
+	router.POST("/query", graphqlHandler())
+	api := router.Group("/api/v1")
+
+	// the jwt middleware
+	var authMiddleware = middleware.GetGinJWTMiddleware()
+
+	// public routes
+	api.POST("/register", validators.RegisterValidator(), handlers.RegisterHandler)
+	api.POST("/login", validators.LoginValidator(), authMiddleware.LoginHandler)
```

Comment:
It would be better to separate out the internal and external API and to have an endpoint that can be used to issue tokens that are specifically API tokens, which are different from the tokens you get from logging in via the web interface.